engine/schema/src/main/java/com/cloud/vm/dao/NicDaoImpl.java

```diff
@@ -214,7 +214,7 @@ public NicVO findByNetworkIdAndMacAddress(long networkId, String mac) {
         SearchCriteria<NicVO> sc = AllFieldsSearch.create();
         sc.setParameters("network", networkId);
         sc.setParameters("macAddress", mac);
-        return findOneBy(sc);
+        return findOneIncludingRemovedBy(sc);
```

Comment:
Is this an intended change for this PR? can you please help me understand why we are including removed entries. 
Also there are other many usages of this method, so I see a chance of regression here.